Library/Homebrew/cmd/audit.rb

```diff
@@ -391,7 +391,9 @@ def audit_options
       end
 
       if o.name =~ /^with(out)?-(?:checks?|tests)$/
-        problem "Use '--with#{$1}-test' instead of '--#{o.name}'. Migrate '--#{o.name}' with `deprecated_option`."
+        unless formula.deps.optional.include? Dependency.new("check", [:optional])
```

Comment:
How about:
```ruby
formula.deps.any? { |d| d.name == "check" && (d.optional? || d.recommended?) }
```

Note, the `recommended?` is used to fix the false positive of `--without-check`.